h/services/oauth_validator.py

```diff
@@ -89,6 +122,15 @@ def validate_redirect_uri(self, client_id, redirect_uri, request, *args, **kwarg
             return (client.redirect_uri == redirect_uri)
         return False
 
+    def validate_refresh_token(self, refresh_token, client, request, *args, **kwargs):
```

Comment:
For security critical pieces like this, I think it would be good to ensure we document what exactly it is you're validating. For example:

```python
"""
Validate a supplied refresh token.

Check that the refresh token supplied with an access token request a) exists, 
b) has not expired, and c) is associated with the client identified in the request.
If we return True from this function, we can assume that it is safe to issue a 
new access token to the requesting client.
"""
```